intake.py

```diff
@@ -488,7 +503,11 @@ def api_intake_lock_dataset(ctx, path, dataset_ids):
     # check permissions - datamanager only
     parts = path.split('/')
     group = parts[3]
+
+    # grp-intake-FOO => grp-datamanager-FOO
     datamanager_group = group.replace("-intake-", "-datamanager-", 1)
```

Comment:
- This code fragment is duplicated a few times in this file. Could be a candidate for extracting a function.
- When extracting  a function, it would also be useful to do some input validation, e.g. something like
```
if group.startswith("grp-intake-"):
   return group.replace("-intake-", "-datamanager-", 1)
elif group.startswith("intake-"):
   return group.replace("intake-", "grp-datamanager-", 1)
else:
   raise ValueError("Unexpected intake group format for group "+group)
```
This would also take care of weird edge cases (e.g. "intake-intake-youth")